pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java

```diff
@@ -571,6 +577,40 @@ private List<PinotTaskConfig> createPinotTaskConfigs(List<SegmentZKMetadata> sel
     return pinotTaskConfigs;
   }
 
+  private Map<String, String> getPushTaskConfig(Map<String, String> batchConfigMap, String downloadUrls) {
+    try {
+      String[] downloadURLList = downloadUrls.split(MinionConstants.SEGMENT_NAME_SEPARATOR);
+      if (downloadURLList.length > 0) {
+        String downloadUrl = downloadURLList[0];
+        URI downloadURI = URI.create(downloadUrl);
+        URI outputDirURI = null;
+        if (!downloadURI.getScheme().contentEquals("http")) {
+          String outputDir = downloadUrl.substring(0, downloadUrl.lastIndexOf("/"));
+          outputDirURI = URI.create(outputDir);
+        }
+        String pushMode = IngestionConfigUtils.getPushMode(batchConfigMap);
+
+        Map<String, String> singleFileGenerationTaskConfig = new HashMap<>(batchConfigMap);
+        if (outputDirURI != null) {
+          URI outputSegmentDirURI = SegmentGenerationUtils.getRelativeOutputPath(
+              outputDirURI, downloadURI, outputDirURI);
+          singleFileGenerationTaskConfig.put(
+              BatchConfigProperties.OUTPUT_SEGMENT_DIR_URI, outputSegmentDirURI.toString());
+        }
+        if ((outputDirURI == null) || (pushMode == null)) {
```

Comment:
When push mode is MetadataPush, we should set the default location `data_dir/tablename` when OUTPUT_SEGMENT_DIR_URI is not set by users. Most of the time, users will not need to set it.